src/Integration.Vsix/SonarLintDaemon/SonarLintDaemonInstaller.xaml.cs

```diff
@@ -31,49 +32,100 @@ namespace SonarLint.VisualStudio.Integration.Vsix
     /// </summary>
     public partial class SonarLintDaemonInstaller : Window
     {
+        private ISonarLintSettings settings;
+        private ISonarLintDaemon daemon;
+
+        private volatile bool canceled = false;
+
+        private Action callback;
+
         public SonarLintDaemonInstaller()
         {
             InitializeComponent();
         }
 
         private void Window_ContentRendered(object sender, EventArgs args)
         {
-            // TODO rewrite using Task.Run, see: https://blog.stephencleary.com/2013/05/taskrun-vs-backgroundworker-intro.html
-            BackgroundWorker worker = new BackgroundWorker();
-            worker.WorkerReportsProgress = true;
-            worker.DoWork += worker_DoWork;
-            worker.RunWorkerCompleted += Worker_RunWorkerCompleted;
-            worker.RunWorkerAsync();
+            Daemon.Install(DownloadProgressChanged, DownloadFileCompleted);
+        }
+
+        private void DownloadProgressChanged(object sender, DownloadProgressChangedEventArgs e)
+        {
+            double bytesIn = double.Parse(e.BytesReceived.ToString());
+            double totalBytes = double.Parse(e.TotalBytesToReceive.ToString());
+            double percentage = bytesIn / totalBytes * 100;
+
+            ProgressBar.Value = (int) Math.Truncate(percentage);
         }
 
-        void worker_DoWork(object sender, DoWorkEventArgs eventArgs)
+        private void DownloadFileCompleted(object sender, AsyncCompletedEventArgs e)
         {
-            try
+            if (e.Error != null)
             {
-                var daemon = ServiceProvider.GlobalProvider.GetMefService<ISonarLintDaemon>();
-                daemon.Install();
-                daemon.Start();
-                ServiceProvider.GlobalProvider.GetMefService<ISonarLintSettings>().SkipActivateMoreDialog = true;
+                var ex = e.Error;
+                var message = string.Format("Failed to activate JavaScript support: {0}", ex.Message);
```

Comment:
You can use the string interpolation here it is easier to read.
`var message = $"Failed to activate JavaScript support: {ex.Message}"`